lib/puppet/parser/functions/docker_run_flags.rb

```diff
@@ -5,17 +5,27 @@
 # docker_run_flags.rb
 #
 module Puppet::Parser::Functions
+  newfunction(:'docker::escape', type: :rvalue) do |args|
+    subject = args[0]
+
+    if self['facts'] && self['facts']['os']['family'] == 'windows'
+      call_function('powershell_escape', subject)
+    else
+      subject.shellescape
```

Comment:
For this to be more readable, maybe remove `require 'shellwords'` above and change this to:

```suggestion
      call_function('shellescape', subject)
```